nodeJS/express/controllers.md

```diff
@@ -108,75 +108,99 @@ function myMiddleware(req, res, next) {
 
 The naming conventions for these controllers are usually based on the route they will be attached to e.g. `GET` route -> `getSomething`, `POST` route -> `createSomething`, `DELETE` route -> `deleteSomething`, etc. Nonetheless, there is no fixed rule since Express is not opinionated. It will always be based on you or someone else's conventions, and the requirements of the function.
 
-Let's define our first controller! Create a folder called `controllers` at the root of your Express project. Then create a `userController.js` file within it:
+You should already have your application setup in-place from the previous lesson. Previously, we've defined a route for `/authors/:authorId` to get the author by their ID, let's now define a controller for that route. But to make the example produce something, let's first create a mock database with a function that we can use to retrieve an author by their ID.
+
+Create the following file at the root of the project, and add the following code:
+
+```javascript
+// db.js
+
+const authors = [
+  { id: 1, name: "Bryan" },
+  { id: 2, name: "Christian" },
+  { id: 3, name: "Jason" },
+];
+
+const getAuthorById = async (authorId) => {
+  return authors.find(author => author.id === authorId);
+};
+
+module.exports = { getAuthorById };
+```
+
+The file name, content, and location aren't important here, they only serve to illustrate that we're using a database query function to retrieve an author by their ID.
+
+Then for the controller, create the following file:
 
 ```javascript
-// user controller file - controllers/userController.js
+// controllers/authorController.js
 
-const getUserById = async (req, res) => {
-  const userId = req.params.id;
+const { getAuthorById } = require("../db");
 
-  const user = await someDBQueryToGetUser(userId);
+const getAuthorById = async (req, res) => {
+  const { authorId } = req.params;
 
-  if (!user) {
-    res.status(404).send("User not found");
+  const author = await getAuthorById(authorId);
+
+  if (!author) {
+    res.status(404).send("Author not found");
     return;
   }
 
-  res.send(`User found: ${user.name}`);
+  res.send(author);
 };
+
+module.exports = { getAuthorById };
 ```
 
-In this example, the `getUserById` function is a controller that handles a specific action related to retrieving a user by their ID. You'll use this controller by importing it into file where routes are defined, and using it like so `router.get("/user/:id", getUserById)`. Let's break down what's happening in this controller:
+In this example, the `getAuthorById` function is a controller that handles a specific action related to retrieving an author by their ID. We'll use this controller by importing it into the file where we define our routes, and using it like this: `router.get("/:authorId", getAuthorById)`. Let's break down what's happening in this controller:
```

Comment:
We should probably do 2 things:
- Move the inline code block to a standalone fenced code block
- In that example, refer to the appropriate router file path (`routes/authorsRouter.js`)

Since I named the authors router  `authorsRouter` in the Routes lesson, we should probable change this example to `authorsRouter.get("/:authorId", getAuthorById)`